parquet-hadoop/src/main/java/org/apache/parquet/crypto/CryptoPropertiesFactory.java

```diff
@@ -51,20 +71,36 @@ public static CryptoPropertiesFactory loadFactory(Configuration conf) throws IOE
 
     try {
       CryptoPropertiesFactory cryptoFactory = (CryptoPropertiesFactory)cryptoPropertiesFactoryClass.newInstance();
-      cryptoFactory.initialize(conf);
       return cryptoFactory;
     } catch (InstantiationException | IllegalAccessException e) {
       throw new BadConfigurationException("could not instantiate CryptoPropertiesFactory class: "
         + cryptoPropertiesFactoryClass, e);
     }
   }
 
-  public abstract void initialize(Configuration hadoopConfig);
-
-  public abstract FileEncryptionProperties getFileEncryptionProperties(
-    Configuration fileHadoopConfig, Path tempFilePath,
-    WriteContext fileWriteContext)  throws IOException;
+  /**
+   * Get FileEncryptionProperties object which is created by the implementation of this interface. Please see
+   * the unit test (TBD) for example
+   *
+   * @param fileHadoopConfig Configuration that is used to pass the needed information, e.g. KMS uri
+   * @param tempFilePath File path of the parquet file being written.
+   *                     Can be used for AAD prefix creation, key material management, etc.
+   *                     Implementations must not presume the path is permanent,
+   *                     as the file can be moved or renamed later
+   * @param fileWriteContext WriteContext to provide information like schema to build the FileEncryptionProperties
+   * @return
+   * @throws IOException
```

Comment:
javadoc generation does not like having these tags empty. Please write something to them as well.

Reply:
Sounds good